atc-router_test.go

```diff
@@ -56,3 +64,44 @@ func Test_GetFields(t *testing.T) {
 	require.NoError(t, err)
 	require.ElementsMatch(t, []any{"http.path", "tcp.port"}, fields)
 }
+
+func Test_splitByNulls(t *testing.T) {
+	for _, test := range []struct {
+		name string
+		buf  []byte
+		n    int
+		out  []string
+	}{
+		{name: "empty input", buf: nil, out: []string{}},
+		{name: "empty input, n:1", buf: nil, n: 1, out: []string{""}},
+		{name: "single string", buf: []byte("one\x00"), n: 1, out: []string{"one"}},
+		{name: "single string, n:0", buf: []byte("one\x00"), n: 0, out: []string{}},
+		{name: "single string, n:2", buf: []byte("one\x00"), n: 2, out: []string{"", "one"}},
+		{name: "unterminated", buf: []byte("one"), n: 1, out: []string{""}},
+		{name: "two, unterminated", buf: []byte("one\x00two"), n: 2, out: []string{"", "one"}},
+		{name: "consecutive separator 3", buf: []byte("one\x00\x00two\x00"), n: 3, out: []string{"", "one", "two"}},
+		{name: "consecutive separator 2", buf: []byte("one\x00\x00two\x00"), n: 2, out: []string{"", "one"}},
```

Comment:
Does it make sense to remove the strings with length 0 before returning the strings array in the function?

Reply:
i did that at first, since i didn't like having the ""s at the start (because they're "leftovers", so should be at the end if anywhere), but found a bit more important to keep the length matching the `n` parameter.

in any case, that shouldn't happen in real world; it's only on slightly malformed input (consecutive or missing separators, wrong requested number of fields, etc)  and since the '\0's are not "separators" but "string terminators", then consecutive '\0\0' is really an empty string.